src/main/kotlin/com/sourcegraph/cody/config/CodyAuthenticationManager.kt

```diff
@@ -114,6 +116,16 @@ class CodyAuthenticationManager :
           override fun windowActivated(e: WindowEvent?) {
             super.windowActivated(e)
             ApplicationManager.getApplication().executeOnPooledThread { getAuthenticationState() }
+            CodyAgentService.withAgent(project) { agent: CodyAgent ->
```

Comment:
Can we extract this listener somewhere an then subscribe to it if we want in places we need?
Doing general state notifications inside `CodyAccountManager` looks weird...